collection/migrate.go

```diff
@@ -42,6 +43,42 @@ func MigrateRepoStoreToLocalCollectionSet(ctx context.Context, s Set, r repo.Rep
 				datasets[i].MetaTitle = ds.Meta.Title
 			}
 		}
+
+		ulog, err := book.UserDatasetBranchesLog(ctx, datasets[i].InitID)
+		if err != nil {
+			log.Errorf("can't get user log for dataset %s, %s", datasets[i].InitID, err)
+			continue
+		}
+		dlog, err := ulog.Log(datasets[i].InitID)
+		if err != nil {
+			log.Errorf("can't get dataset log for dataset %s, %s", datasets[i].InitID, err)
+			continue
+		}
+		if len(dlog.Logs) < 0 {
+			log.Errorf("no branch logs for dataset log %s, %s", datasets[i].InitID, err)
+			continue
+		}
+		blog := dlog.Logs[0]
+		commitCount := 0
+		runCount := 0
+		mostRecentRunRecorded := false
+		for _, op := range blog.Ops {
+			if op.Model == logbook.CommitModel {
+				commitCount++
+				continue
+			}
```

Comment:
this is close, but you need to account for removed and amended versions as well. Here's an example, but instead of constructing new VersionInfos, you'd be adjusting commitCount to decrement on remove type actions: 

https://github.com/qri-io/qri/blob/4e63b2f8ecafab531370a8ea45232e64df45002c/logbook/logbook.go#L1234-L1257

Reply:
oh nice thank you! I completely forgot that remove ops are also model type `CommitModel`.